addons/pos_hr/models/pos_session.py

```diff
@@ -7,6 +7,11 @@
 class PosSession(models.Model):
     _inherit = 'pos.session'
 
+    def _pos_data_process(self, loaded_data):
+        super()._pos_data_process(loaded_data)
+        if self.config_id.module_pos_hr:
+            loaded_data['employee_by_id'] = {employee['id']: employee for employee in loaded_data['hr.employee']}
```

Comment:
It feels like the `employee_by_id` is enough to be sent to client. We can just delete `hr.employee` from the loaded data and in the client, any use of it, let's just use `employee_by_id`. Unless of course, the ordering of `hr.employee`s is important.

Reply:
added in todolist